crates/sncast/src/helpers/braavos.rs

```diff
@@ -77,12 +80,58 @@ where
         self.block_id
     }
 
-    async fn sign_deployment(
+    async fn sign_deployment_v1(
         &self,
-        deployment: &RawAccountDeployment,
+        deployment: &RawAccountDeploymentV1,
     ) -> Result<Vec<FieldElement>, Self::SignError> {
         let tx_hash =
-            PreparedAccountDeployment::from_raw(deployment.clone(), self).transaction_hash();
+            PreparedAccountDeploymentV1::from_raw(deployment.clone(), self).transaction_hash();
+
+        let signature = self.signer.sign_hash(&tx_hash).await?;
+
+        let mut aux_data = vec![
```

Comment:
Can you please leave a comment here (or a link to the doc + comment) why it's done like this? I know, you know but probably somebody later can be confused by this

Reply:
I could not find any documentation on that, except for the comment in the code of `braavos-account-contract`. You can find the link [here](https://github.com/myBraavos/braavos-account-cairo/blob/6efdfd597bb051e99c79a512fccd14ee2523c898/src/presets/braavos_account.cairo#L104). However, if we look into the deserialize [code](https://github.com/myBraavos/braavos-account-cairo/blob/6efdfd597bb051e99c79a512fccd14ee2523c898/src/presets/braavos_account.cairo#L141), it explicitly explains why the code data needs to be structured that way.